datastore.go

```diff
@@ -274,6 +275,16 @@ func (t *txn) Query(q dsq.Query) (dsq.Results, error) {
 	opt := badger.DefaultIteratorOptions
 	opt.PrefetchValues = !q.KeysOnly
 
+Loop:
+	for _, o := range q.Orders {
```

Comment:
Same as in https://github.com/ipfs/go-ds-leveldb/pull/22#discussion_r246099431, we have to apply these sequentially. Again, we'll probably want to work backwards till we find *either* an `OrderByKey` or an `OrderByKeyDescending`.